pkg/reaper/pdbreaper/types.go

```diff
@@ -0,0 +1,137 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pdbreaper
+
+import (
+	"fmt"
+
+	"github.com/keikoproj/governor/pkg/reaper/common"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// Args is the argument struct for pdb-reaper
+type Args struct {
+	K8sConfigPath     string
+	DryRun            bool
+	LocalMode         bool
+	ReapMisconfigured bool
+	ReapMultiple      bool
+	ReapCrashLoop     bool
+}
+
+// ReaperContext holds the context of the pdb-reaper and target cluster
+type ReaperContext struct {
+	KubernetesClient                           kubernetes.Interface
+	KubernetesConfigPath                       string
+	DryRun                                     bool
+	LocalMode                                  bool
+	ReapMisconfigured                          bool
+	ReapMultiple                               bool
+	ReapCrashLoop                              bool
+	ReapablePodDisruptionBudgets               []policyv1beta1.PodDisruptionBudget
```

Comment:
You're copying a lot of the same data into multiple maps/slices. It would be more efficient if you stored here just the namespace & name in this `ReapablePodDisruptionBudgets` slice as that's all that's required for the Delete call or use pointers `*policyv1beta1.PodDisruptionBudget`

Reply:
Can you elaborate?
Data is copied a single time when it's added to the slice of PDBs when it's detected

Reply:
It appears you store the full object `policyv1beta1.PodDisruptionBudget` in multiple places, maybe in a map you have the full object then you append it to the slice of `ReapablePodDisruptionBudgets`, instead of copying by value multiple places you can use refs or maybe have a slice of namespace/name of pdb b/c that slice is only used for a delete call where `namespace` and `name` are required only.